byterun/caml/mlvalues.h

```diff
@@ -214,8 +228,8 @@ typedef opcode_t * code_t;
 
 /* Another special case: objects */
 #define Object_tag 248
-#define Class_val(val) Field((val), 0)
-#define Oid_val(val) Long_val(Field((val), 1))
+#define Class_val(val) Field_imm((val), 0)
+#define Oid_val(val) Long_val(Field_imm((val), 1))
```

Comment:
You should check the code carefully, but at least one of these seems to be mutable.  (See `stdlib/camlinternalOO.ml` at the top, for example.)